src/ast/mod.rs

```diff
@@ -1247,6 +1247,14 @@ pub enum Statement {
         /// deleted along with the dropped table
         purge: bool,
     },
+    /// DROP Function
+    DropFunction {
+        if_exists: bool,
+        /// One or more function to drop
+        func_desc: Vec<DropFunctionDesc>,
+        cascade: bool,
```

Comment:
@zidaye maybe an enum here would be better (or an option with a boolean inside).

Here you allow the state "cascade + restrict", which doesn't make sense at all.

Reply:
@AugustoFKL done. very detailed comment